Lib/fontTools/varLib/instancer.py

```diff
@@ -1099,6 +1117,379 @@ def pruningUnusedNames(varfont):
+    #
+    # AxisValues remaining after task 1.3:
+    # axisValues = [
+    #     {"name": "Medium", "axis": "wght", "value": 500},
+    #     {"name": "Normal", "axis": "wdth", "value": 100, "flags": 0x2}
+    # ]
+    #
+    # AxisValues remaining after completing all 1.x tasks:
+    # axisValues = [{"name": "Medium", "axis": "wght", "value": 500}]
+    # The Normal AxisValue is removed because it has the
+    # Elidable_AXIS_VALUE_NAME flag set.
+    #
+    # # AxisValues after separating into two groups in task 2.1:
+    # ribbiAxisValues = []
+    # nonRibbiAxisValues = [{"name": "Medium", "axis": "wght", "value": 500}]
+    #
+    # # Names created from AxisValues in task 2.2a for Win US English platform:
+    # subFamilyName = ""
+    # typoSubFamilyName = "Medium"
+    #
+    # NameRecords updated in task 2.2b for Win US English platform:
+    # NameID 1 familyName: "Open Sans" --> "Open Sans Medium"
+    # NameID 2 subFamilyName: "Regular" --> "Regular"
+    # NameID 3 Unique font identifier: "3.000;GOOG;OpenSans-Regular" --> \
+    #     "3.000;GOOG;OpenSans-Medium"
+    # NameID 4 Full font name: "Open Sans Regular" --> "Open Sans Medium"
+    # NameID 6 PostScript name: "OpenSans-Regular" --> "OpenSans-Medium"
+    # NameID 16 Typographic Family name: None --> "Open Sans"
+    # NameID 17 Typographic Subfamily name: None --> "Medium"
+    #
+    # Notes on name table record updates:
+    # - Typographic names have been added since Medium is a non-Ribbi name.
+    # - Neither the before or after name records include the Width AxisValue
+    #   names because the "Normal" AxisValue has the
+    #   Elidable_AXIS_VALUE_NAME flag set.
+    #   If we instantiate the same font but pin the wdth axis to 75,
+    #   the "Condensed" AxisValue will be included.
+    # - For info regarding how RIBBI and non-RIBBI can be constructed see:
+    # https://docs.microsoft.com/en-us/typography/opentype/spec/name#name-ids
+
+    if "STAT" not in varfont:
+        raise ValueError("Cannot update name table since there is no STAT table.")
+    stat = varfont["STAT"].table
+    if not stat.AxisValueArray:
+        raise ValueError("Cannot update name table since there are no STAT Axis Values")
+    fvar = varfont["fvar"]
+
+    # The updated name table must reflect the new 'zero origin' of the font.
+    # If we're instantiating a partial font, we will populate the unpinned
+    # axes with their default axis values.
+    fvarDefaults = {a.axisTag: a.defaultValue for a in fvar.axes}
+    defaultAxisCoords = deepcopy(axisLimits)
+    for axisTag, val in fvarDefaults.items():
+        if axisTag not in defaultAxisCoords or isinstance(
+            defaultAxisCoords[axisTag], AxisRange
+        ):
+            defaultAxisCoords[axisTag] = val
+
+    axisValueTables = axisValuesFromAxisLimits(stat, defaultAxisCoords)
+    checkAxisValuesExist(stat, axisValueTables, defaultAxisCoords)
+
+    # Remove axis Values which have ELIDABLE_AXIS_VALUE_NAME flag set.
```

Comment:
saying "remove" implies we're modifying the axis values, whereas we're just filtering them, including some excluding others

```suggestion
    # Ignore axis Values which have ELIDABLE_AXIS_VALUE_NAME flag set.
```